### PR TITLE
Xcode Project Refactor

### PR DESCRIPTION
**Pull request by cnoon**
This PR updates the Xcode project to use workspace linking instead of nested Xcode projects. This allows clients using submodules to use a different version of Alamofire to link to AlamofireImage in their own workspace.

> This is something I've been meaning to change for quite some time. We originally couldn't do this because we had issues in I believe Xcode 10 where `xcodebuild` was linking the incorrect Alamofire.framework version. Sometimes it would link `Alamofire watchOS` to `AlamofireImage iOS` and such.

I've also verified this change still builds safely with Carthage.

### CODE REVIEW

File: AlamofireImage.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2153,44 +2146,100 @@
...
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				MTL_ENABLE_DEBUG_INFO = INCLUDE_SOURCE;
+				MTL_FAST_MATH = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = "org.alamofire.iOS-Example";
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				PROVISIONING_PROFILE_SPECIFIER = "";
+				SWIFT_ACTIVE_COMPILATION_CONDITIONS = DEBUG;
+				SWIFT_OBJC_BRIDGING_HEADER = "Example/Supporting Files/iOS-Example-Bridging-Header.h";
+				SWIFT_VERSION = 4.2;
```
Comment (jshier) [inline AlamofireImage.xcodeproj/project.pbxproj]: Can you delete this setting so from the example target, so we can keep just the root project toggle?

File: AlamofireImage.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2153,44 +2146,100 @@
...
+				CLANG_ENABLE_OBJC_WEAK = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_STYLE = Automatic;
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = "";
+				GCC_C_LANGUAGE_STANDARD = gnu11;
+				INFOPLIST_FILE = "Example/Supporting Files/Info.plist";
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (jshier) [inline AlamofireImage.xcodeproj/project.pbxproj]: Should set this back down to iOS 8 for now.

File: AlamofireImage.xcodeproj/project.pbxproj (hunk 1 above)
Comment (cnoon) [reply AlamofireImage.xcodeproj/project.pbxproj]: Good catch! That was not intentional. Will set back.

File: AlamofireImage.xcodeproj/project.pbxproj (hunk 2 above)
Comment (cnoon) [reply AlamofireImage.xcodeproj/project.pbxproj]: Absolutely, that was not intended either.

**Pull request merged**